app/src/test/java/com/fstyle/structure_android/screen/main/MainPresenterTest.java

```diff
@@ -39,12 +43,112 @@
 
     @Before
     public void setUp() throws Exception {
-        MockitoAnnotations.initMocks(this);
         mMainPresenter.setView(mMainView);
         mMainPresenter.setSchedulerProvider(mSchedulerProvider);
     }
 
     @Test
+    public void validateKeywordInput_WhenEmptyThenInvokesMessageNoEmptyAndReturnFalse()
+            throws Exception {
+        // Give
+        boolean expect = false;
+        String msg = "Must not empty";
+
+        // When
+        when(mValidator.validateValueNonEmpty(anyString())).thenReturn(msg);
+
+        // Then
+        boolean actual = mMainPresenter.validateKeywordInput(anyString());
+
+        verify(mMainView).onInvalidKeyWord(msg);
+        Assert.assertEquals(expect, actual);
+    }
+
+    @Test
+    public void validateKeywordInput_WhenInputNGWordThenInvokesMessageNGWordAndReturnFalse()
+            throws Exception {
+        // Give
+        boolean expect = false;
+        String msg = "There are unusable characters";
+
+        // When
+        when(mValidator.validateValueNonEmpty(anyString())).thenReturn("");
+        when(mValidator.validateNGWord(anyString())).thenReturn(msg);
+
+        // Then
+        boolean actual = mMainPresenter.validateKeywordInput(anyString());
+
+        verify(mMainView).onInvalidKeyWord(msg);
+        Assert.assertEquals(expect, actual);
+    }
+
+    @Test
+    public void validateKeywordInput_WhenInputKeywordValidThenReturnTrue() throws Exception {
+        // Give
+        boolean expect = true;
+
+        // When
+        when(mValidator.validateValueNonEmpty(anyString())).thenReturn("");
+        when(mValidator.validateNGWord(anyString())).thenReturn("");
+
+        // Then
+        boolean actual = mMainPresenter.validateKeywordInput(anyString());
+
+        Assert.assertEquals(expect, actual);
+    }
+
+    @Test
+    public void validateLimitNumberInput_WhenEmptyThenInvokesMessageNoEmptyAndReturnFalse()
+            throws Exception {
+        // Give
+        boolean expect = false;
+        String msg = "Must not empty";
+
+        // When
+        when(mValidator.validateValueNonEmpty(anyString())).thenReturn(msg);
+
+        // Then
+        boolean actual = mMainPresenter.validateLimitNumberInput(anyString());
+
+        verify(mMainView).onInvalidLimitNumber(msg);
+        Assert.assertEquals(expect, actual);
+    }
+
+    @Test
+    public void
+    validateLimitNumberInput_WhenInputInvalidNumberThenInvokesMessageNGWordAndReturnFalse()
```

Comment:
ERROR: 'validateLimitNumberInput_WhenInputInvalidNumberThenInvokesMessageNGWordAndReturnFalse' have incorrect indentation level 4, expected level should be 8.